src/profile.c

```diff
@@ -246,11 +251,8 @@ static void fcloseMaybeCompressed(struct ProfileState* ps)
 ** as approriate, and then pass through to the true function
 */
 
-// This function checks if we have ever printed out the id of stat
-static inline UInt getFilenameId(Stat stat)
+static inline UInt getFilenameIdOfBody(Obj body)
```

Comment:
The comment right before this function now fits there even less than it did before... move it?

Reply:
Also, I find it really surprising that a function with this name has a major side-effect (it prints some JSON to `profileState.Stream` under certain conditions). It would be nice if there was at least a comment in there which addressed this. I'd find it even better if this printing did *not* happen in there, but instead was moved to the callers, resp. to a separate function `outputFileIdIfNecessary` or so.

Of course this problem predates this PR, so I am no expecting it to be resolved in this PR, I can also take care of it myself after this PR is merged.

Reply:
I'm happy to rename, but I quite like the printing happens in here, as it means you can't forget to print out the id before using it.